doc/glossary.rst

```diff
@@ -205,6 +205,23 @@ General Concepts
         exceptional behaviours on the estimator using semantic :term:`estimator
         tags`.
 
+    cross-fitting
+    cross fitting
+        A resampling method that iteratively partitions data into complementary
+        'train' and 'estimation' subsets. The 'estimation' subset may be used
+        to estimate a parameter or predict a target. The outputs from
+        each iteration are combined and used in a downstream prcess, generally
+        to avoid bias.
```

Comment:
I don't find the "train" and "estimation" names very clear because one could argue that training and estimating are similar. Maybe it would be helpful to speak in terms of first and second stage as follows:

```suggestion
        A resampling method that iteratively partitions data into mutually exclusive
        subsets to fit a two stage estimator. The second stage is fit on predictions
        of the first stage computed on a subset of data not seen when training the
        first stage. The objective is to avoid having any overfitting in the first
        stage introduce a bias in the training data distribution of the second stage.
```

Reply:
'mutually exclusive' is much clearer and more explicit than 'complementary', have used this in the cross validation entry too, thanks!
Is `TargetEncoder` technically a 2 stage estimator? There isn't really a second stage, it just outputs the transformed X values for use by another estimator...? I have thus tried to make it more general and re-worded. Not entirely happy though and open to changes. Also I may have mis-understood some aspects!

Reply:
`TargetEncoder` is the first stage and the HistGradientBoosting model (or anything else) you train on the output of `TargetEncoder` is the second stage.

For stacking and calibration with use meta-estimator, whereas for `TargetEncoder` we use a transformer pipeline and implement the cross-fitting only in the `fit_transform` of the first stage transformer.

Reply:
Thanks, yes that is my understanding too.

Now I realise my mis-understanding is over what 'two stage estimator' means. Initially I took it to mean that a single estimator that performs two stages, which none of the estimators do (`CalibratedClassifierCV` technically takes a `estimator` argument which does the first stage). Whereas you meant 'two stage estimator' as an estimator that is designed to perform a 2 stage process. What about changing the wording to be something like: 
"to fit a two stage estimator chain" or "to fit an estimator in a two stage process/model" ??

Reply:
ping @ogrisel , would either suggestion be suitable?